nameserver/status.go

```diff
@@ -0,0 +1,48 @@
+package nameserver
+
+type Status struct {
+	Domain  string
+	Address string
+	TTL     uint32
+	Entries []EntryStatus
+}
+
+type EntryStatus struct {
+	Hostname    string
+	Origin      string
+	ContainerID string
+	Address     string
+	Version     int
+	Tombstone   int64
+}
+
+func NewStatus(ns *Nameserver, dnsServer *DNSServer) *Status {
+	if dnsServer == nil {
+		return nil
+	}
+
+	return &Status{
+		dnsServer.domain,
+		dnsServer.address,
+		dnsServer.ttl,
+		NewEntryStatusSlice(ns)}
+}
+
+func NewEntryStatusSlice(ns *Nameserver) []EntryStatus {
+	ns.RLock()
+	defer ns.RUnlock()
+
+	var slice []EntryStatus
+	for _, entry := range ns.entries {
+		slice = append(slice,
+			EntryStatus{
```

Comment:
Can you put this on the previous line, or does gofmt produce something ghastly then?